docs/src/pages/components/data-grid/selection/selection.md

```diff
@@ -62,8 +62,16 @@ In the demo below only rows with quantity above 50000 can be selected:
 
 ## Controlled selection
 
+Use the `selectionModel` prop to control the grid selection. Each time the `selectionModel` changes the `onSelectionModelChange` callback is called with the new selection model value.
+
 {{"demo": "pages/components/data-grid/selection/ControlledSelectionGrid.js", "bg": "inline"}}
 
+### Controlled selection and server-side pagination
+
+Using controlled selection when `paginationMode="server"` can result in the `DataGrid` clearing the provided `selectionModel` on page change. Depending on your implementation of the server-side pagination, when the page changes and there are no rows in the grid with ids equal to the `GridRowId`s provided to the `selectionModel` the grid will clear the `selectionModel`. To prevent this either save the `selectionModel` and restore it later or append the newly loaded grid rows to the existing rows.
```

Comment:
I tried to make it less technical.

```suggestion
Using the controlled selection with `paginationMode="server"` may result in selected rows being lost when the page is changed.
This happens because the grid cross-checks with the `rows` prop and only calls `onSelectionModelChange` with existing row IDs.
Depending on your server-side implementation, when the page changes and the new value for the `rows` prop does not include previously selected rows, the grid will call `onSelectionModelChange` with an empty value.
To prevent this unwanted behavior, there are two ways:

- Save the `selectionModel` **before the page is changed** and restore it later
- Append the newly loaded rows to the existing rows

The following demo shows to implement the second solution:
```

Reply:
Looking again and https://github.com/mui-org/material-ui-x/pull/2602#issuecomment-919045908 is right, the Select All is broke. Adding `checkboxSelectionVisibleOnly` doesn't help and it still selects the appended rows. We can try the first solution which is a bit more hacky or we investigate why the prop didn't work.

Reply:
True, but I would treat that as a separate issue and investigate it in an independent PR. This one should be to just add clarity around the relationship between the server-side pagination and the selection model.